metrics/stats.go

```diff
@@ -0,0 +1,161 @@
+package metrics
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Stats struct {
+	Time int64 `json:"time"`
+	// runtime
+	GoVersion    string `json:"go_version"`
+	GoOs         string `json:"go_os"`
+	GoArch       string `json:"go_arch"`
+	CpuNum       int    `json:"cpu_num"`
+	GoroutineNum int    `json:"goroutine_num"`
+	Gomaxprocs   int    `json:"gomaxprocs"`
+	CgoCallNum   int64  `json:"cgo_call_num"`
+	// memory
+	MemoryAlloc      uint64 `json:"memory_alloc"`
+	MemoryTotalAlloc uint64 `json:"memory_total_alloc"`
+	MemorySys        uint64 `json:"memory_sys"`
+	MemoryLookups    uint64 `json:"memory_lookups"`
+	MemoryMallocs    uint64 `json:"memory_mallocs"`
+	MemoryFrees      uint64 `json:"memory_frees"`
+	// stack
+	StackInUse uint64 `json:"memory_stack"`
+	// heap
+	HeapAlloc    uint64 `json:"heap_alloc"`
+	HeapSys      uint64 `json:"heap_sys"`
+	HeapIdle     uint64 `json:"heap_idle"`
+	HeapInuse    uint64 `json:"heap_inuse"`
+	HeapReleased uint64 `json:"heap_released"`
+	HeapObjects  uint64 `json:"heap_objects"`
+	// garbage collection
+	GcNext           uint64    `json:"gc_next"`
+	GcLast           uint64    `json:"gc_last"`
+	GcNum            uint32    `json:"gc_num"`
+	GcPerSecond      float64   `json:"gc_per_second"`
+	GcPausePerSecond float64   `json:"gc_pause_per_second"`
+	GcPause          []float64 `json:"gc_pause"`
+	// Connections
+	Connections int64 `json:"connections"`
+}
+
+type safeTime struct {
+	time.Time
+	sync.RWMutex
+}
+
+func (t *safeTime) get() time.Time {
+	t.RLock()
+	defer t.RUnlock()
+	return t.Time
+}
+
+func (t *safeTime) set(tm time.Time) {
+	t.Lock()
+	defer t.Unlock()
+	t.Time = tm
+}
+
+var nsInMs float64 = float64(time.Millisecond)
```

Comment:
__[golint]__
* [golint] should omit type float64 from declaration of var nsInMs; it will be inferred from the right-hand side

<!-- Comment by [SideCI](https://sideci.com) -->